src/hotspot/cpu/aarch64/c1_MacroAssembler_aarch64.cpp

```diff
@@ -81,7 +81,7 @@ int C1_MacroAssembler::lock_object(Register hdr, Register obj, Register disp_hdr
   }
 
   if (LockingMode == LM_LIGHTWEIGHT) {
-    lightweight_lock(obj, hdr, temp, rscratch2, slow_case);
+    lightweight_lock(disp_hdr, obj, hdr, temp, rscratch2, slow_case);
```

Comment:
Given the declaration:
```
void MacroAssembler::lightweight_lock(Register basic_lock, Register obj, Register t1, Register t2, Register t3, Label& slow) 
```
it looks odd to pass `disp_hdr` here - is that variable just mis-named?

Reply:
Yeah, because the BasicLock and display header / metadata in BasicLock are all on the same address so they have been used interchangeably in c1,c2 and the interpreter. It should probably be fixed. But maybe in a separate PR.